website/docs/plugin/best-practices/index.mdx

```diff
@@ -0,0 +1,65 @@
+---
+page_title: Plugin Development - Best Practices
+description: >-
+  Patterns that ensure a consistent user experience, including naming and versioning.
+---
+
+# Terraform Plugin Best Practices
+
+A key feature of Terraform is its plugin system, which separates the details of
+specific vendor APIs from the shared logic for managing state, managing configuration,
+and providing a safe plan and apply lifecycle. Plugins are responsible
+for the  implementation of functionality for provisioning resources for a
+specific cloud provider, allowing each provider to fully support its unique
+resources and lifecycles and not settling for the lowest common denominator
+across all provider resources of that type (virtual machines, networks,
+configuration management systems, et. al). While each provider is unique, over
+the years, we accumulated recommended patterns that help ensure a consistent user experience when using Terraform for any given provider. This page describes best practices that generally apply to most
+Providers, with a brief description of each, and link to read more. 
+
+## Handling Sensitive Data
+
+In [Handling Sensitive Data][sensitive-state], we discuss the recommended approaches for managing sensitive state in Terraform.
+
+## Provider Design Principles
+
+Over time, the Terraform development community has gained large amounts of experience across a wide breadth of Application Programming Interfaces (APIs) supported by Terraform and its Software Development Kits (SDKs). The [Provider Design Principles][design-principles] highlight the most prolific patterns when designing Terraform providers and associated resources, which guide HashiCorp design decisions.
+
+## Interacting with Providers
+
+Terraform's providers are a substantial amount of code, and occasionally it makes sense to depend on their functionality. [Interacting with Providers][interacting-with-providers] describes different ways to interface with Terraform's providers outside of the context of the Terraform CLI.
+
+## Naming
+
+[Naming][naming] resources, data sources, and attributes in plugins is how plugin authors
+expose their functionality to operators and using patterns common to other plugins
+lays the foundation for a good user experience.
+
+## Provider Code
+
+In [Provider Code][provider-code], we describe the supported programming languages and HashiCorp libraries that can be used to develop Terraform Providers.
+
+## Versioning and Changelog
+
+Terraform development serves two distinct audiences: those writing plugin code
+and those implementing them. By clearly and consistently allowing operators to
+easily understand changes in plugin implementation via version numbering and
+documenting those changes, a trust is formed between the two audiences. In
+[Versioning and Changelog][versioning] we cover some guidelines when deciding
+release versions and how to relay changes through documentation.
+
+[sdkv2]: /terraform/plugin/sdkv2
```

Comment:
Nit: A lot of our other documentation uses inline Markdown links, if you'd prefer to switch to that.

Reply:
done ⚡